### PR TITLE
[ticket/15351] Makes confirm_works in a router context (app.php)

### PR DESCRIPTION
**Pull request by Nicofuma**
PHPBB3-15351

Checklist:

- [ ] Correct branch: master for new features; 3.2.x, 3.1.x for fixes
- [ ] Tests pass
- [ ] Code follows coding guidelines: [master / 3.2.x](https://area51.phpbb.com/docs/master/coding-guidelines.html), [3.1.x](https://area51.phpbb.com/docs/31x/coding-guidelines.html)
- [ ] Commit follows commit message [format](https://wiki.phpbb.com/Git#Commit_Messages)

Tracker ticket (set the ticket ID to **your ticket ID**):

https://tracker.phpbb.com/browse/PHPBB3-15351

### CODE REVIEW

File: phpBB/phpbb/path_helper.php [hunk 1]
```diff
@@ -496,4 +496,17 @@ public function get_valid_page($page, $mod_rewrite = false)
 
 		return $page;
 	}
+
+	/**
+	 * Tells if the router is currently in use (if the current page is a route or note)
```
Comment (Elsensee) [inline phpBB/phpbb/path_helper.php]: Typo

File: phpBB/phpbb/path_helper.php [hunk 2]
```diff
@@ -496,4 +496,17 @@ public function get_valid_page($page, $mod_rewrite = false)
...
+	 *
+	 * @return bool
+	 */
+	public function is_router_used()
+	{
+		// Script name URI (e.g. phpBB/app.php)
+		$script_name = $this->symfony_request->getScriptName();
+
+		return basename($script_name) === 'app.' . $this->php_ext;
```
Comment (Elsensee) [inline phpBB/phpbb/path_helper.php]: Does this also work when the .htaccess redirection is in place?
Bugs that are connected to this are nasty.

Comment (Nicofuma) [reply phpBB/phpbb/path_helper.php]: yes it works. The script name should always be defined

Comment (Elsensee) [reply phpBB/phpbb/path_helper.php]: Surely it's always defined but does it also always point to the app.php when routing through it but is redirected by .htaccess?

Comment (Nicofuma) [reply phpBB/phpbb/path_helper.php]: yep

**Pull request merged**